frontend/apps/reader/modules/readerbookmark.lua

```diff
@@ -215,7 +215,38 @@ function ReaderBookmark:gotoBookmark(pn_or_xp)
     end
 end
 
+-- This function adds "chapter" property to highlights already saved in the document
+function ReaderBookmark:updateHighlightsIfNeeded()
+    local version = self.ui.doc_settings:readSetting("bookmarks_version") or 0
+    if version >= 20200511 then
+        return
+    end
+
+    for page, highlights in pairs(self.view.highlight.saved) do
+        for _, highlight in pairs(highlights) do
+            local pg_or_xp = self.ui.document.info.has_pages and
+                    page or highlight.pos0
+            local chapter_name = self.ui.toc:getTocTitleByPage(pg_or_xp)
+            highlight.chapter = chapter_name
+        end
+    end
+
+    for _, bookmark in ipairs(self.bookmarks) do
+        if bookmark.pos0 then
+            local pg_or_xp = self.ui.document.info.has_pages and
+                    bookmark.page or bookmark.pos0
+                local chapter_name = self.ui.toc:getTocTitleByPage(pg_or_xp)
+            bookmark.chapter = chapter_name
+        elseif bookmark.page then -- dogear bookmark
+            local chapter_name = self.ui.toc:getTocTitleByPage(bookmark.page)
+            bookmark.chapter = chapter_name
+        end
+    end
+    self.ui.doc_settings:saveSetting("bookmarks_version", 20200511)
```

Comment:
Can you just update this `20200511` (here and above) to something closer to the day this will be merged, let's say `20200615` if a release happen this weekend :)
So we get a more accurate date for when this migration has happened.

Reply:
done :+1: